src/main/java/randoop/output/FailingTestFilter.java

```diff
@@ -0,0 +1,277 @@
+   * @throws BugInRandoopException if {@code status} contains output for a failure not involving a
+   *     Randoop generated test method
+   */
+  private String commentFailingAssertions(
+      String classname, String javaCode, RunCommand.Status status) {
+    // JUnit4 writes to standard out, check this doesn't change.
+    assert status.errorOutputLines.isEmpty()
+        : "Expecting JUnit to write to standard out, but found output on standard error";
+
+    /* Iterator to move through JUnit output. */
+    Iterator<String> lineIterator = status.standardOutputLines.iterator();
+
+    /*
+     * First, find the message that indicates the number of failures in the run.
+     */
+
+    Pair<String, String> failureCountMatch = readUntilMatch(lineIterator, FAILURE_MESSAGE_PATTERN);
+    int totalFailures = Integer.parseInt(failureCountMatch.b);
+    assert totalFailures > 0 : "JUnit has non-zero exit status, but no failure found";
+
+    /*
+     * Then read the rest of the file to find each failure.
+     * The standard runner gives a numbered list with each entry matching the following pattern:
+     */
+    Pattern failureHeaderPattern =
+        Pattern.compile("\\d+\\)\\s+(" + ID_STRING + ")\\(" + ID_STRING + "\\)");
+
+    /*
+     * Split Java code text so that we can match the line number for the assertion with the code.
+     * Use same line break as used to write test class file.
+     */
+    String[] javaCodeLines = javaCode.split(Globals.lineSep);
+
+    int failureCount = 0;
+    while (failureCount < totalFailures) {
+      /*
+       * Read until beginning of failure
+       */
+      Pair<String, String> failureHeaderMatch = readUntilMatch(lineIterator, failureHeaderPattern);
+      String line = failureHeaderMatch.a;
+      String methodName = failureHeaderMatch.b;
+
+      /*
+       * If the method name in the failure message is not a test method, throw an exception.
+       */
+      if (!methodName.matches(GenTests.TEST_METHOD_NAME_PREFIX + "\\d+")) {
+        if (line.contains("initializationError")) {
+          throw new BugInRandoopException(
+              "Check configuration of test environment: "
+                  + "initialization error of test in flaky-test filter");
+        } else {
+          throw new BugInRandoopException("Unexpected failure in flaky-test filter: " + methodName);
+        }
+      }
+
+      /*
+       * Search for the stacktrace entry corresponding to the test method, and capture the line
+       * number.
+       */
+      Pattern linePattern =
+          Pattern.compile(
+              String.format(
+                  "\\s+at\\s+%s\\.%s\\(%s\\.java:(\\d+)\\)", classname, methodName, classname));
+
+      Pair<String, String> failureLineMatch = readUntilMatch(lineIterator, linePattern);
+      int lineNumber = Integer.parseInt(failureLineMatch.b);
+      if (lineNumber < 1 && lineNumber < javaCodeLines.length + 1) {
```

Comment:
This looks wrong.  It should be `||` instead of `&&`, it should be `>` instead of `<`, and why is the `+1` needed?